iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -408,12 +409,12 @@ private void commitTable(FileIO io, ExecutorService executor, OutputTable output
 
     Table table = null;
     String branchName = null;
-
+    Expression filterExpr = Expressions.alwaysTrue();
     for (JobContext jobContext : outputTable.jobContexts) {
       JobConf conf = jobContext.getJobConf();
       table = Optional.ofNullable(table).orElse(Catalogs.loadTable(conf, catalogProperties));
       branchName = conf.get(InputFormatConfig.OUTPUT_TABLE_SNAPSHOT_REF);
-
+      filterExpr = Expressions.and(filterExpr, HiveIcebergStorageHandler.getQueryFilter(conf));
```

Comment:
not sure what is happening here.
filterExpr is true, and we do a AND with that, why?
we go inside ``getQueryFilter``, in that again we have a ``true`` and then we do a AND
```
  Expression expr = Expressions.alwaysTrue();
    try {
      expr = Expressions.and(expr,
```

Final it looks like TRUE AND (TRUE AND EXPR)
why is it not just EXPR?

Reply:
I have fixed getQueryFilter, it removes the one extra AND step.

Reply:
why do we need this in a jobContext loop?

Reply:
````
if (filterExpr != null) {
    filterExpr = SessionStateUtil.getResource(x.getJobConf(), InputFormatConfig.QUERY_FILTERS)
    .orElse(Expressions.alwaysTrue())
}
````